README.md

```diff
@@ -30,71 +30,88 @@ The Getting Started instructions can be found [here](GettingStarted.md).
 
   1. Composable atomic swaps.
   2. Users maintain custody of their assets at all times.
-  3. Naturally concurrent and gets more concurrent the more users there are. No batchers are required.
-  4. Liquidity naturally spreads to all trading pairs instead of being siloed into specific trading pairs.
-  5. There is no impermanent loss.
-  6. ADA is all you need to interact with the DEX.
-  7. Upgradability can happen democratically.
-  8. Easy to integrate into any frontend.
+  3. Naturally concurrent and **scales *with* the number of users**. No batchers required.
+  4. Liquidity spreads organically to all trading pairs instead of being siloed into specific trading pairs/liquidity pools.
+  5. No impermanent loss for individual swaps (Of course, there is still inventory risk for *traders*, which is not the same as IL).
+  6. No secondary tokens - ADA is all you need to interact with the DEX.
+  7. Democratic upgradability
+  8. Relatively straightforward integration with existing frontends (i.e. wallets)
 
-In addition to these nice properties, the novel use of *Beacon Tokens* can be generalized to any application on Cardano.
+Additionally, the novel use of *Beacon Tokens* can be generalized to many applications on Cardano, not just DEXes.
 
 ---
-## Personal Contracts
+
+## Specification
+
+### Personal Contracts
 In order for users to maintain full delegation control of their assets, it is required for user assets to be kept siloed. The reason for this is that, on Cardano, all ADA at an address is either delegated entirely or not at all (atomic delegation is not possible). So for a user to maintain delegation control of their assets, only that user's assets should be at the address he/she is using.
 
-A cardano address is made up of both a payment credential and a staking credential. As long as the staking credential is unique to that user, that user will maintain full delegation control for that address. With this in mind, every user is given the same spending script to use for the address' payment credential and are required to use their own staking credential (either a pubkey or a staking script). To force the use of a staking credential, the DEX is designed so that it is not possible to mint a beacon token (discussed later) to an address without a staking credential.
+A cardano address is made up of both a payment credential and a staking credential. As long as the staking credential is unique to that user, full delegation control over the address is maintained. With this in mind, every user is given the same spending script to use for the address' payment credential and are required to use their own staking credential (either a pubkey or a staking script). To force the use of a staking credential, the DEX is designed such that it is not possible to mint a valid beacon token (discussed later) to an address without a staking credential.
```

Comment:
Drop the "valid" here for "mint a *valid* beacon token" since it isn't possible to mint anything if the address doesn't have a staking credential.

Reply:
like at all ever for any token?

Reply:
At all. The minting policy only allows minting one kind of token: the token with the Policy ID for the minting script and the empty token name. Tokens with any other token name are not mintable. Further, the newly minted token *must* go to cardano-swaps address with a valid staking credential. If the address doesn't have one, the minting policy is guaranteed to fail. Therefore nothing can be minted.

In short, either the proper beacon token is minted or nothing is minted.